extensions/iceberg/src/main/java/io/deephaven/iceberg/util/IcebergCatalogAdapter.java

```diff
@@ -549,6 +575,13 @@ private Table readTableInternal(
         return result;
     }
 
+    static Table readTableNoSnapshot(
+            @NotNull final Schema schema,
+            @NotNull final org.apache.iceberg.Table table,
+            @NotNull final IcebergInstructions userInstructions) {
+        return TableTools.newTable(tableDefinition(schema, table, userInstructions, -1));
```

Comment:
This works for static tables, but for refreshing tables I believe we'll need to return `PartitionAwareSourceTable` with zero partitions (or the `IcebergTable` equivalent) in order to populate data with discovered files.